tests/unit/rule_strings/pipeline.txt

```diff
@@ -0,0 +1,58 @@
+import linker.utilities.validation_utils as validation_utils
+rule all:
+    input:
+        final_output=['{snake_dir}/result.parquet'],
+        validation='{snake_dir}/input_validations/final_validator'
+rule:
+    name: "results_validator"
+    input: ['{snake_dir}/result.parquet']
+    output: touch("{snake_dir}/input_validations/final_validator")
+    localrule: True         
+    message: "Validating results input"
+    run:
+        for f in input:
+            validation_utils.validate_dummy_file(f)
+rule:
+    name: "step_1_python_pandas_validator"
+    input: ['{test_dir}/input_data1/file1.csv', '{test_dir}/input_data2/file2.csv']
```

Comment:
Why are there two input files?

Reply:
good question

Reply:
It's because there are two files in the default config input_data yaml, and `get_input_files()` just indiscriminately picks up all the input_data files for the first implementation in the pipeline:
`    def get_input_files(self, implementation: Implementation, results_dir: Path) -> list:
        idx = self.implementation_indices[implementation.name]
        if idx == 0:
            return [str(file) for file in self.config.input_data]`
This is consistent with the old run method, but obviously it will need to change as there are input files that go to different steps.